application/classes/Ushahidi/Repository/Post/Media.php

```diff
@@ -14,9 +14,56 @@
 
 class Ushahidi_Repository_Post_Media extends Ushahidi_Repository_Post_Value
 {
+
+	// Ushahidi_Repository
+	public function getEntity(Array $data = null)
+	{
+		$data['value'] = $data['o_filename'];
```

Comment:
forcing 'value' to be filename here could have consequences in client. Need to check

Reply:
might be best to let it be an id and change the getAllForPost

Reply:
the problem with this is that we use the id for getting media from the server so we can't do this except for csv